projects/control-service/projects/pipelines_control_service/src/main/java/com/vmware/taurus/service/KubernetesService.java

```diff
@@ -1040,7 +1078,7 @@ public void updateCronJob(
       Map<String, String> jobLabels,
       List<String> imagePullSecrets)
       throws ApiException {
-    if (getK8sSupportsV1CronJob()) {
+    if (readV1CronJob(name).isPresent()) {
```

Comment:
```suggestion
    if (getK8sSupportsV1CronJob() && readV1CronJob(name).isPresent()) {
```

Reply:
If for some reason there are cron jobs using v1 and v1beta1 in the same namespace, wouldn't checking the flag cause unexpected behaviour?

Reply:
The flag indicates if the v1 is supported or not. In my opinion, it will be strange to see the v1 job if the flag is set to false.

Reply:
@tozka What’s your view on that?

Reply:
This was done in the context of https://github.com/vmware/versatile-data-kit/pull/1580#issuecomment-1413626260

Reply:
Paul's comment makes sense. In case of a revert, if a job is created with the new API, the control service will 'forget' it. However, this would require an additional Kubernetes API call, which is not optimal. Therefore, the decision to make it forward-compatible or not should be based on the expected performance overhead it would add.

It is worth noting that any new users would be using Kubernetes 1.21, where the V1 API is supported, and we plan to switch k8sSupportsV1CronJob to true by default soon. As a result, I would be okay with an approach that is not forward-compatible (i.e., without the extra API call).

On a separate note, I wish we had automated testing to measure how many API calls (and other resources) we use.